vm/package/is_build_required.rs

```diff
@@ -39,34 +39,32 @@ impl<N: Network> Package<N> {
             Err(_) => return true,
         };
 
-        // Initialize a boolean indicator if we need to build the circuit.
-        let mut is_complete = true;
-
         // Check if the program ID in the manifest matches the program ID in the AVM file.
-        if avm_file.program().id() == &self.program_id {
-            // Retrieve the main program.
-            let program = self.program();
+        if avm_file.program().id() != &self.program_id {
```

Comment:
Are we sure this is correct? If the program ID doesn't match, I believe we shouldn't be building because it is the wrong program.

Reply:
I see, I may be lacking some context here, I thought this was something like a lighter version of the "has this program changed" check. Is there a legitimate use case for skip building on different program id, or is it more like an invalid condition where the cli should throw an error?

Reply:
The latter. Your intuition is correct here. This is like another sanity checking spot, which should never technically be hit . I suppose we could just remove the check?

Reply:
Yeah I think so. I'll take a look at the sorrounding code and run some checks to be sure, but if it looks good I'll just remove it.

Reply:
I tried @facundoolano's changes in the roulette project removing this check.
When the program IDs match, the result is the expected:
```$ aleo build

⏳ Compiling 'bets.aleo'...

 • Loaded universal setup (in 1610 ms)
 • Built 'psd_hash' (in 6252 ms)
 • Built 'mint_casino_token_record' (in 11657 ms)
 • Built 'make_bet' (in 26166 ms)
 • Built 'psd_bits_mod' (in 5980 ms)

✅ Built 'bets.aleo' (in "/Users/ncontinanza/Projects/Entropy1729/aleo_roulette/circuits/bets")
```

When the IDs do not match (I edited the ID in the program.json file), we get a descriptive error:
```$ aleo build

⚠️  The program ID from `program.json` does not match in '/Users/ncontinanza/Projects/Entropy1729/aleo_roulette/circuits/bets/main.aleo'```

Reply:
Nice we can remove the check then